narwhals/expression.py

```diff
@@ -1417,6 +1418,54 @@ def is_last_distinct(self) -> Expr:
         """
         return self.__class__(lambda plx: self._call(plx).is_last_distinct())
 
+    def quantile(
+        self,
+        quantile: float,
+        interpolation: Literal["nearest", "higher", "lower", "midpoint", "linear"],
+    ) -> Expr:
+        r"""Get quantile value.
+
+        Arguments:
+            quantile : float
+                Quantile between 0.0 and 1.0.
+            interpolation : {'nearest', 'higher', 'lower', 'midpoint', 'linear'}
+                Interpolation method.
+
+        Examples:
+            >>> import narwhals as nw
+            >>> import pandas as pd
+            >>> import polars as pl
+            >>> data = {"a": list(range(50)), "b": list(range(50, 100))}
+            >>> df_pd = pd.DataFrame(data)
+            >>> df_pl = pl.DataFrame(data)
+
+            Let's define a dataframe-agnostic function:
+
+            >>> def func(df_any):
+            ...     df = nw.from_native(df_any)
+            ...     result = df.select(nw.col("a", "b").quantile(0.5, interpolation="nearest"))
+            ...     return nw.to_native(result)
+
+            We can then pass either pandas or Polars to `func`:
+
+            >>> func(df_pd)  # doctest: +NORMALIZE_WHITESPACE
+                a   b
+            0  24  74
+
+            >>> func(df_pl)  # doctest: +NORMALIZE_WHITESPACE
+            shape: (1, 2)
+            ┌──────┬──────┐
+            │ a    ┆ b    │
+            │ ---  ┆ ---  │
+            │ f64  ┆ f64  │
+            ╞══════╪══════╡
+            │ 25.0 ┆ 75.0 │
+            └──────┴──────┘
```

Comment:
i'm a little surprised that they output different results, have you checked why?

Reply:
True! The quantile falls in the middle of two numbers. numpy and pandas implementation of `method="nearest"` seem to round down while rust/polars to round up? 
Should we use a more stable example?

Reply:
yeah maybe 'linear' in the examples, and then add a note explaining that there may be some slight implementation differences when rounding?